### PR TITLE
make User Is Not Authorized error more informative

### PR DESCRIPTION
**Pull request by britneywright**
Add record or zone contact information to User is Not Authorized error message in batch change/DNS Change.

### CODE REVIEW

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala [hunk 1]
```diff
@@ -486,7 +494,19 @@ class BatchChangeValidations(
...
+      .leftMap(_ =>
+        ownerGroupId match {
+          case Some(id) =>
+            UserIsNotAuthorizedError(authPrincipal.signedInUser.userName, id, "", "record")
+          case None =>
+            UserIsNotAuthorizedError(
+              authPrincipal.signedInUser.userName,
+              input.zone.adminGroupId,
+              input.zone.email,
+              "zone")
```
Comment (pauljamescleary) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeValidations.scala]: As opposed to a hardcoded string, instead to use a Sum type (enum) for the value

**Comment by britneywright** [comment]
> @britneywright can we also have tests added for user should not have access to modify a record that they are the owner group of in a private zone and is returning the correct error message? Our access validations don't allow this, but just want to make sure that the error messaging reflects this.

This sounds like it should be a separate issue and PR.

**Comment by britneywright** [comment]
> Not sure I follow, @britneywright - that should be this PR since before it was the same error message before and we're now changing the error message and how the fields are populated.

Added a record to test in the existing `test_a_recordtype_update_delete_checks` func test.

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 2]
```diff
@@ -126,7 +131,40 @@ class BatchChangeService(
...
+            val group = groups.find(_.id == e.ownerGroupId)
+            val updatedError = UserIsNotAuthorizedError(
+              e.userName,
+              e.ownerGroupId,
+              e.ownerType,
+              group.map(_.email),
+              group.map(_.name))
+            logger.error(updatedError.message)
+            updatedError
+          case a =>
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: `e` is preferable here for errors.

File: modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala [hunk 3]
```diff
@@ -126,7 +131,40 @@ class BatchChangeService(
...
+            logger.error(a.message)
+            a
+        }.invalid
+      case Valid(a) => a.validNel
```
Comment (mitruly) [inline modules/api/src/main/scala/vinyldns/api/domain/batch/BatchChangeService.scala]: This can just be `case valid => valid`.

**Pull request merged**